org.eclipse.lsp4e/src/org/eclipse/lsp4e/ConnectDocumentToLanguageServerSetupParticipant.java

```diff
@@ -83,6 +85,11 @@ protected IStatus run(IProgressMonitor monitor) {
 					});
```

Comment:
If `LanguageServiceAccessor.getLanguageServers()` fails and thenRun() is not executed, would the job still be removed from SUBMITTED_JOBS?

Reply:
Fair point - I've correct this to do it in the main body of the job run method in all cases - once we get to that point it's too late to cancel so there's no point in keeping that entry for the test teardown.

I don't think it's easy to make the tests bulletproof - my original change was just an attempt to mitigate a major source of noise/flakiness. But test hooks shouldn't cause resource leaks in production...